functions/helpers/addCustomerExternalId.private.ts

```diff
@@ -12,6 +13,12 @@ type EnvVars = {
 
 const TASK_CREATED_EVENT = 'task.created';
 
+const logAndReturnError = (taskSid: TaskInstance['sid'], workspaceSid: EnvVars['TWILIO_WORKSPACE_SID'], step: 'fetch' | 'update', errorInstance: unknown) => {
+  const errorMessage = `Error at addCustomerExternalId: task with sid ${taskSid} does not exists in workspace ${workspaceSid} when trying to ${step} it.`;
+  console.error(errorMessage, errorInstance);
```

Comment:
Will this create the duplicate error messages that we are getting rid of elsewhere? Or are we returning a 200 in Line 19, so it's still only one error in the logs?

This does feel like a much more clean and clear message than we had before.

Reply:
This will catch the error in the fetch/update task steps, logging the error so the error reporter is raised once, but we'll still return an error message to `taskRouterCallback`, instead than causing it to throw. Hence, `taskRouterCallback` will just complete it's execution (not throwing), but the error will be reported and will appear in the monitor once because of this `console.error`.
The only place that could cause `taskRouterCallback` to really throw, is if the task's attributes object can't be parsed, which I don't think is even possible, but we can also wrap it in a try/catch if that makes sense.

Reply:
Sounds good, thank you! I agree that checking for parseable attributes doesn't seem necessary.